### DISCUSSION TITLE
New Login Concept

### DISCUSSION THREAD

**Discussion by stevew00dy** [sunflower-land/sunflower-land#1173] [Ideas]
I propose to move the current login system from a web3 wallet address to an email / password version. 

The reasons are as follows:

1, Mass Adoption
As we move towards educating the general public into Sunflower Land and scale to 1mil users, many will be coming to the game without the awareness of Crypto and so I feel it's important that we introduce them to the game without the need to have anything setup. A simple Email confirmation and password would allow for a user to access and play the game but not withdraw anything from their farm. 

2, Additional Device
For so many Sunflower Land should be a game that is picked up and then put back down, awaiting times for tasks to complete but we are not always at a PC, sometimes people are mobile and there should be a way that a user can access both the desktop and mobile version of the game. This is currently limited for those who use a hardware wallet for additional protection. 

3, The need to verify every 24 hours
I feel that with other games, they do not require this checkin with the wallet, it's a step we could look to remove so that users can enjoy an uninterrupted session without needing to confirm a signature every day.

**Comment by NotKeen** [comment]
I like the idea. Things so consider:

- Minting the farm / syncing all progress to chain would be delayed to whenever the new player gets around to connecting their wallet and paying the minting fee (if any) I guess? This would unlock all features that were locked before like withdrawals? minting NFTs?
 On-chain activities in general for sure
- Maybe consider skipping mail and just do it via Discord logins? Debatable tho, not everyone may like Discord, mail is more neutral. Discord login might be easier to use tho

- About playing with a hardware device, maybe let all critical transactions like withdrawals, big NFT mints or trading be confirmed by signature, but otherwise try to minimize the need for signing any transactions. Ofc this kinda turns the whole Goblin Village approach on its head, so not sure how feasible this is

- Maybe instead of needing to sync to restock seeds etc there could be other ways, like time limits (enforced server-side) that players have to wait until more seeds and other stock becomes available

- Right now funding relies a lot on syncing fees (to my knowledge). If they become less, would have to think more about alternative funding models.

**Comment by vrillx** [comment]
Why not have both methods available ?

Make login in with email /pass as primary ...first thing new players see and underneath a ''sign in with metamask '' secondary option ..

Metamask is very convenient for crypto users , as a login feature for its one click login ability, removing it may alienate exactly the people we want to play the game crypto users ...

**Reply by IvanSkazkin** [reply]
> Почему бы не использовать оба метода?
> 
> Войдите в систему с помощью электронной почты / пароля в качестве основного ... первое, что видят новые игроки, и под дополнительной опцией «войти с помощью метамаски» ..
> 
> Метамаска очень удобна для крипто-пользователей, так как функция входа в систему позволяет войти в систему одним щелчком мыши, ее удаление может оттолкнуть именно тех людей, которых мы хотим играть в крипто-игры пользователей ...

I think so too

**Comment by Dukler** [comment]
I don't actually like the concept of moving it completely from web3 out to email version, I would keep an in-between those 2, kind of like what axie does, where you can login in with email & password, but you cant actually do anything blockchain related because you would need to sign a transaction with your wallet. Since with an email password is easier to get hacked because of information leaks from 3rd party software and all that jazz (I know this would be user's fault, because bad password administration). But I really like the layer of security the blockchain provides having no signing capability from the email & password version seems appealing to me. Maybe this problem I'm mentioning could be mitigated with extra steps like 2FA. So I don't know I'm a bit on the fence here, maybe having both systems?

**Comment by stevew00dy** [comment]
My thoughts are that when someone first joins the game - we need to assume they are not web3. But we want to convert them. So for this they would need traditional web2 user / pass to get started with basic gameplay features. 

then as they learn, they would be required to connect a wallet. This would mint their farm and enable the web3 elements of the game. 

I think at this stage it would be good to have both options. They can login using user / pass but only do web2 gameplay (this would allow mobile gameplay without the metamask app and wouldn't affect hard wallet users) and then when they want to do web3, they sign a transaction. 

To give them the option for both at login makes sense (user / pass or login with wallet - click button (access metamask, wallet connect or coinbase)

**Comment by JcEii** [comment]
I think this will make it easier for other users of SFL. Two-factor authentication is another suggestion to verify the player to avoid hacking of the account. Metamask is required once the player withdrew his/her sfl, sync the game to restock the supplies of items and minting NFTs. Email is an alternative way to login to the game. On the other hand, it is better to have face detection verification too (two-factor authentication and face detection with government or valid ID; these two will make the account strong and help to avoid multi-account).

**Comment by ShiningMinions** [comment]
I agree with your stated problem (inconvenience), but I disagree with your proposed solution (email/password login). Adding an email/password solution adds yet another layer of centralization and takes away from the true end-vision of the team (which is complete decentralization). We should be working towards the end goal, not go backwards. Pioneering this new web3 territory is what makes this game awesome. "Gamifying" web3 for complete web3 noobs - that's half the fun of this game! Forgetting this web3 root will make SFL look like any other game, and therefore will become nothing special.

So really, what I think what we **should** be talking about, is how do we make the game more convenient for the average player, but without sacrificing decentralization and its web3 roots?

Ideas:
- Use other wallets like coinbase (I know it's already a WIP)
- Allow other chains
- Allow other wallets besides metamask
- Somehow get Polygon way easier to setup in metaMask
- etc.

**Reply by Aminbtdds** [reply]
I agree with you pointing out centralization/decentralization issue.
When we try to make everything decentralized(everything that is possible), making starting of game with email, doesn't imply that idea, imo, we can have a tutorial stage which would educate newcomers with zero web3 knowledge, and make them ready for the game.
If i could educate my 65 years old dad(he is anesthesiologist but i tried 3 years to learn him crypto wallet concept, and i failed before sfl), perhaps proper tutorial will do the job for those who spend 15min. to learn things.
I guess in the terms of development and mass adoption in any kind of business, we should consider "what makes my project unique and what is the core appeal?!", if we lost those answer on the way of developments, we literally shooting ourselves in the foot.
Thus far, main appeal for me in our game was, trying to be decentralized, being community driven(specially being heard by team), play to own concept.
From 1.5k beta testers to 145k beta testers, imho, we made sacrifices in all aspects more or less, for the well being of project we should address those sacrifices and prevent further ones.

P.s.
Btw, My dad called goblins, green rabits🤣🤣🤣🌻

**Comment by NotKeen** [comment]
Maybe just make the beginning stage of the game (like a demo) cookie based or whatever (savegame ID stored in browser), not even a login needed. Just to get the new players interested in the gameplay. Then at some point tell them to create a wallet and make it as easy / gamified as possible.

**Reply by ShiningMinions** [reply]
I think that's a very solid idea.

**Reply by Aminbtdds** [reply]
👌👏👏👏
